ts/src/pro/kucoin.ts

```diff
@@ -1055,6 +1056,10 @@ export default class kucoin extends kucoinRest {
         //    }
         //
         const data = this.safeString (message, 'data', '');
+        if (data === 'token is expired') {
+            const type = client.url.substring (client.url.indexOf ('connectId=') + 10);
```

Comment:
@sc0Vu did you test this in py or PHP? Because I don't think substring is transpilable